src/psij/__init__.py

```diff
@@ -31,44 +31,66 @@
 
 
 class _PluginType:
-    def __init__(self, name: str, registration_method: Callable[[Type[T]], None]):
+    def __init__(self, name: str, registration_method: Callable[[_Descriptor, str], None]):
         self.name = name
         self.registration_method = registration_method
-        self.package = ['psij', name]
 
 
-TYPES = [_PluginType('executors', JobExecutor.register_executor),
-         _PluginType('launchers', Launcher.register_launcher)]
+PACKAGE = ['psij-descriptors']
 
 
-def _load_plugins(path: str, type: _PluginType) -> None:
-    for mod in pkgutil.iter_modules(path=[path]):
-        if not mod.ispkg and not mod.name[0] == '_':
-            im = importlib.import_module('psij.{}.{}'.format(type.name, mod.name))
-            var_name = '__PSI_J_{}__'.format(type.name.upper())
-            if hasattr(im, var_name):
-                classes = getattr(im, var_name)
-                logger.debug('Found module "{}" with classes {}'.format(mod.name, classes))
-                for cls in classes:
-                    logger.debug('Registering {}'.format(cls))
-                type.registration_method(cls)
+TYPES = [_PluginType('executors', JobExecutor.register_executor),
+         _PluginType('launchers', Launcher.register_launcher)]
 
 
-def _find_plugins(path: str, type: _PluginType, ix: int) -> None:
-    if ix == len(type.package):
-        _load_plugins(path, type)
-    else:
-        for mod in pkgutil.iter_modules(path=[path]):
-            if mod.ispkg and mod.name == type.package[ix]:
-                _find_plugins(path + os.path.sep + mod.name, type, ix + 1)
+def _load_plugins(root: str, full_path: str, mod: pkgutil.ModuleInfo) -> None:
+    if not mod.ispkg and not mod.name[0] == '_':
```

Comment:
what's excluded by this `_prefix` clause?

Reply:
It's a remnant from when executors were there and you maybe wanted to have a non-executor in the same package. I don't think it's needed now given that we have a special package for only descriptors.